addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/ExternalRedirectScanRule.java

```diff
@@ -313,7 +315,7 @@ static String getRefreshUrl(String value) {
     }
 
     private static final Pattern LOCATION_PATTERN =
-            Pattern.compile("(?i)\\s*url\\s*=\\s*[\"']?([^'\"]*)[\"']?");
+            Pattern.compile("(?i)^\\s*url\\s*=\\s*[\"']?([^'\"]*)[\"']?");
```

Comment:
Added leading anchor so that Refresh payloads aren't incorrectly caught.

Reply:
Sounds like we don't need the specific refresh pattern.

Reply:
The refresh pattern account for the leading digit. Without the anchor here a refresh can look like a location because location is a subset or substring.

Reply:
I understand that, but we are not using the leading digits for anything.

Reply:
I'm not seeing your vision. Yes they're very similar but if this one needs to be anchored in order to not be mismatched then they're still different.

Reply:
Mismatched from what? In which case is the anchor needed?

Reply:
When the payload is a refresh payload.

Reply:
That makes sense, thank you.